source/extensions/path/pattern_template_lib/pattern_template.h

```diff
@@ -0,0 +1,58 @@
+#pragma once
+
+#include <string>
+
+#include "source/extensions/path/pattern_template_lib/pattern_template_internal.h"
+#include "source/extensions/path/pattern_template_lib/proto/pattern_template_rewrite_segments.pb.h"
+
+#include "absl/status/statusor.h"
+#include "absl/strings/string_view.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace PatternTemplate {
+
+enum class RewriteStringKind { Variable, Literal };
+
+struct RewritePatternSegment {
```

Comment:
What's the difference between this struct, RewriteSegment in pattern_template_internal.h and PatternTemplateRewriteSegments from the .proto? I'm mildly surprised that we need 3 such similarly name structs.

Reply:
On further look this is duplicated. I've removed the struct completely